third_party/move/move-model/src/ty.rs

```diff
@@ -1227,14 +1227,21 @@ impl Substitution {
                 return Ok(Type::Tuple(self.unify_vec(
                     // Note for tuples, we pass on `variance` not `sub_variance`. A shallow
                     // variance type will be effective for the elements of tuples,
-                    // which are treated similar as expression lists in function calls.
+                    // which are treated similar as expression lists in function calls, and allow
+                    // e.g. reference type conversions.
                     context, variance, order, ts1, ts2, "tuples",
                 )?));
             },
             (Type::Fun(a1, r1), Type::Fun(a2, r2)) => {
+                // Same as for tuples, we pass on `variance` not `sub_variance`, allowing
+                // conversion for arguments. We also have contra-variance of arguments:
+                //   |T1|R1 <= |T2|R2  <==>  T1 >= T2 && R1 <= R2
```

Comment:
Is `T1` here the same as `a1` in the code?  Can you use the same variables to avoid confusion?

Reply:
This is very clear as it is, so I keep at is.